src/api/java/com/minecolonies/api/colony/buildings/modules/IDefinesCoreBuildingStatsModule.java

```diff
@@ -0,0 +1,24 @@
+package com.minecolonies.api.colony.buildings.modules;
+
+/**
+ * Interface describing core building stats.
+ * The first core stats module that is found in the building will define the values.
+ */
+public interface IDefinesCoreBuildingStatsModule extends IBuildingModule
+{
+    /**
+     * Get the max building level a module allows.
+     * By default this is 5.
+     * @return the max level.
+     */
+    default int getMaxBuildingLevel()
+    {
+        return 5;
+    }
+
+    /**
+     * Get the max number of inhabitants this module allows.
+     * @return the modules max number of assigned citizens.
+     */
+    int getMaxInhabitants();
```

Comment:
So I thought about this a bit.
I agree that these are indeed the core stats.
But I think we can flush out the API a bit here.
Hit me up on Discord we jump in voice and I will explain what my thoughts here were how we can make this a bit more precise.